ProcessesApi.Tests/V1/Services/ChangeOfNameServiceTests.cs

```diff
@@ -46,11 +47,40 @@ public async Task InitialiseStateToEnterNewNameIfCurrentStateIsNotDefinedAndTrig
             CurrentStateShouldContainCorrectData(process,
                                                  triggerObject,
                                                  ChangeOfNameStates.EnterNewName,
-                                                 new List<string>() { });
+                                                 new List<string>() { ChangeOfNamePermittedTriggers.EnterNewName });
             process.PreviousStates.Should().BeEmpty();
 
             _mockSnsGateway.Verify(g => g.Publish(It.IsAny<EntityEventSns>(), It.IsAny<string>(), It.IsAny<string>()), Times.Exactly(2));
             _lastSnsEvent.EventType.Should().Be(ProcessEventConstants.PROCESS_STARTED_AGAINST_PERSON_EVENT);
         }
+
+        [Fact]
+        public async Task CurrentStateIsUpdatedToNameSubmittedOnEnterNewName()
+        {
+            // Arrange
+            var process = CreateProcessWithCurrentState(ChangeOfNameStates.EnterNewName);
+
+            var newName = "Update";
+            var formData = new Dictionary<string, object>
+            {
+                { ChangeOfNameKeys.FirstName, newName },
+            };
+
+            var triggerObject = CreateProcessTrigger(process,
+                                                     ChangeOfNamePermittedTriggers.EnterNewName,
+                                                     formData);
+
+            // Act
+            await _classUnderTest.Process(triggerObject, process, _token).ConfigureAwait(false);
+
+            // Assert
+            CurrentStateShouldContainCorrectData(process,
+                                                 triggerObject,
+                                                 ChangeOfNameStates.NameSubmitted,
+                                                 new List<string>() { /*TODO: Add next state here */ });
+            process.PreviousStates.LastOrDefault().State.Should().Be(ChangeOfNameStates.EnterNewName);
+            VerifyThatProcessUpdatedEventIsTriggered(ChangeOfNameStates.EnterNewName, ChangeOfNameStates.NameSubmitted);
+        }
+
```

Comment:
Can you add a test that checks that it fails if none of the formdata fields are filled in?